test/e2e/activationscale/activation_scale_test.go

```diff
@@ -0,0 +1,90 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+)
+
+const activationScale = 5
+
+func TestActivationScale(t *testing.T) {
+	t.Parallel()
+
+	ctx := e2e.SetupSvc(t, autoscaling.KPA, autoscaling.Concurrency, 6, 0.7,
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.ActivationScaleKey:        strconv.Itoa(activationScale),
+			autoscaling.InitialScaleAnnotationKey: "0", // avoid having to wait for scale to zero
+		}))
+	test.EnsureTearDown(t, ctx.Clients(), ctx.Names())
+	ctx.SetLogger(t.Logf)
+
+	if err := sendSingleRequest(ctx); err != nil {
+		t.Errorf("%s", err)
+	}
+	if err := checkScaleUp(ctx, activationScale); err != nil {
+		t.Errorf("%s", err)
+	}
+}
+
+func sendSingleRequest(ctx *e2e.TestContext) error {
+	pacer := vegeta.ConstantPacer{
+		Freq: 1,
+		Per:  1 * time.Second,
+	}
+	clients := ctx.Clients()
+	resources := ctx.Resources()
+
+	attacker := vegeta.NewAttacker(vegeta.Timeout(1), vegeta.Workers(1), vegeta.MaxWorkers(1))
+	target, err := e2e.GetVegetaTarget(
+		clients.KubeClient, resources.Route.Status.URL.URL().Hostname(), pkgTest.Flags.IngressEndpoint, test.ServingFlags.ResolvableDomain, "sleep", 500)
+	if err != nil {
+		return fmt.Errorf("error creating vegeta target: %w", err)
+	}
+	attacker.Attack(vegeta.NewStaticTargeter(target), pacer, 0, "load-test")
```

Comment:
We should stop the attacker when the test is done via `t.Cleanup`

https://github.com/tsenart/vegeta/blob/v12.8.4/lib/attack.go#L329